warehouse/accounts/models.py

```diff
@@ -184,6 +184,26 @@ def can_reset_password(self):
             ]
         )
 
+    @property
+    def can_reserve_projects(self) -> bool:
+        if self.is_superuser or self.is_moderator or self.is_psf_staff:
+            # Special users (superusers, moderators, PSF staff) can always reserve projects.
+            return True
+        else:
+            # Ordinary users can reserve projects under the following conditions:
+            #  * Must have a verified primary email
+            #  * Must have 2FA enabled
+            #  * Must be at least 90 days old
+            #  * Must not already have a project name reserved
+            has_already_reserved_project = any(p.is_reserved for p in self.projects)
+            ninety_days_ago = datetime.datetime.now() - datetime.timedelta(days=90)
+            return (
+                self.has_primary_verified_email
+                and self.has_two_factor
+                and self.date_joined <= ninety_days_ago
+                and not has_already_reserved_project
```

Comment:
Flagging for review: do I need to check `User.is_frozen` here as well? I don't think I do since I believe that frozen users can't log in to begin with, but I could be wrong about that.